### ISSUE TITLE
** (FunctionClauseError) no function clause matching in :prim_inet.recv0/3

### ISSUE THREAD

**Issue by nikitosing** [blockscout/blockscout#8491]
after upgrading hackney (https://github.com/blockscout/blockscout/pull/8341). It started to produce a lot of similar errors from different places:
```
2023-09-20T12:38:02.921 fetcher=internal_transaction count=2 [error] Task #PID<0.4009.1> started from Indexer.Fetcher.InternalTransaction terminating
** (FunctionClauseError) no function clause matching in :prim_inet.recv0/3
    (erts 13.2.2.2) :prim_inet.recv0(#Port<0.103>, -1, 600000)
    (hackney 1.18.2) /Users/nikitosing/work/blockscout/blockscout/deps/hackney/src/hackney_response.erl:170: :hackney_response.stream_body_recv/2
    (hackney 1.18.2) /Users/nikitosing/work/blockscout/blockscout/deps/hackney/src/hackney_response.erl:312: :hackney_response.read_body/3
    (hackney 1.18.2) /Users/nikitosing/work/blockscout/blockscout/deps/hackney/src/hackney.erl:520: anonymous fn/2 in :hackney.body/2
    (httpoison 2.1.0) lib/httpoison/base.ex:888: HTTPoison.Base.request/6
    (ethereum_jsonrpc 5.2.2) lib/ethereum_jsonrpc/http/httpoison.ex:12: EthereumJSONRPC.HTTP.HTTPoison.json_rpc/4
    (ethereum_jsonrpc 5.2.2) lib/ethereum_jsonrpc/http.ex:73: EthereumJSONRPC.HTTP.chunked_json_rpc/3
    (ethereum_jsonrpc 5.2.2) lib/ethereum_jsonrpc/request_coordinator.ex:87: anonymous fn/3 in EthereumJSONRPC.RequestCoordinator.perform/4
    (ethereum_jsonrpc 5.2.2) lib/ethereum_jsonrpc/request_coordinator.ex:107: EthereumJSONRPC.RequestCoordinator.trace_request/2
    (ethereum_jsonrpc 5.2.2) lib/ethereum_jsonrpc.ex:441: EthereumJSONRPC.json_rpc/2
    (ethereum_jsonrpc 5.2.2) lib/ethereum_jsonrpc/trace_replay_block_transactions.ex:49: EthereumJSONRPC.TraceReplayBlockTransactions.fetch_block_internal_transactions/3
    (indexer 5.2.2) lib/indexer/fetcher/internal_transaction.ex:115: Indexer.Fetcher.InternalTransaction.run/2
    (elixir 1.14.5) lib/task/supervised.ex:89: Task.Supervised.invoke_mfa/2
    (elixir 1.14.5) lib/task/supervised.ex:34: Task.Supervised.reply/4
    (stdlib 4.3.1.2) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
Function: &Indexer.BufferedTask.log_run/1
    Args: [%{batch: [30063558, 30063559], callback_module: Indexer.Fetcher.InternalTransaction, callback_module_state: [transport: EthereumJSONRPC.HTTP, else: EthereumJSONRPC.IPC, transport_options: [http: EthereumJSONRPC.HTTP.HTTPoison, url: "...", fallback_url: nil, fallback_trace_url: nil, method_to_url: [eth_getBalance: "...", trace_block: "...", trace_replayBlockTransactions: "...", trace_replayTransaction: "..."], http_options: [recv_timeout: 600000, timeout: 600000, hackney: [pool: :ethereum_jsonrpc]]], variant: EthereumJSONRPC.Nethermind], metadata: [fetcher: :internal_transaction]}]
```

```
2023-09-20T12:38:16.488 [error] Task #PID<0.24678.1> started from #PID<0.24655.1> terminating
** (FunctionClauseError) no function clause matching in :prim_inet.recv0/3
    (erts 13.2.2.2) :prim_inet.recv0(#Port<0.125>, -1, 600000)
    (hackney 1.18.2) /Users/nikitosing/work/blockscout/blockscout/deps/hackney/src/hackney_response.erl:170: :hackney_response.stream_body_recv/2
    (hackney 1.18.2) /Users/nikitosing/work/blockscout/blockscout/deps/hackney/src/hackney_response.erl:312: :hackney_response.read_body/3
    (hackney 1.18.2) /Users/nikitosing/work/blockscout/blockscout/deps/hackney/src/hackney.erl:520: anonymous fn/2 in :hackney.body/2
    (httpoison 2.1.0) lib/httpoison/base.ex:888: HTTPoison.Base.request/6
    (ethereum_jsonrpc 5.2.2) lib/ethereum_jsonrpc/http/httpoison.ex:12: EthereumJSONRPC.HTTP.HTTPoison.json_rpc/4
    (ethereum_jsonrpc 5.2.2) lib/ethereum_jsonrpc/http.ex:73: EthereumJSONRPC.HTTP.chunked_json_rpc/3
    (ethereum_jsonrpc 5.2.2) lib/ethereum_jsonrpc/request_coordinator.ex:87: anonymous fn/3 in EthereumJSONRPC.RequestCoordinator.perform/4
    (ethereum_jsonrpc 5.2.2) lib/ethereum_jsonrpc/request_coordinator.ex:107: EthereumJSONRPC.RequestCoordinator.trace_request/2
    (ethereum_jsonrpc 5.2.2) lib/ethereum_jsonrpc.ex:441: EthereumJSONRPC.json_rpc/2
    (ethereum_jsonrpc 5.2.2) lib/ethereum_jsonrpc/receipts.ex:136: EthereumJSONRPC.Receipts.fetch/2
    (elixir 1.14.5) lib/task/supervised.ex:89: Task.Supervised.invoke_mfa/2
    (elixir 1.14.5) lib/task/supervised.ex:34: Task.Supervised.reply/4
    (stdlib 4.3.1.2) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
Function: &:erlang.apply/2
    Args: [#Function<0.128145448/1 in Indexer.Block.Fetcher.Receipts.fetch/2>, [[%{block_hash: "0x7f7e94411aa09e0384db618b3a7342ca9f6aa0071340965cec273959dc885873", block_number: 30063563, from_address_hash: "0xbec4d4983ec4a05a7c3fd1a50b60837f6aaa0205", gas: 1777186, gas_price: 2000000008, hash: "0xcfc92b6c4fd8f4ad8ec94289a1218a41b8023dbcc4f116e2171d175fb49b8835", index: 0, input: "0x37cb2a5e000000000000000000000000f75d379a0730a2cc258786a39fb6aba5ab39af5d", max_fee_per_gas: 2000000016, max_priority_fee_per_gas: 2000000000, nonce: 20219, r: 93813922829229407661418328336918448143251938503113049615011700171792845543642, s: 25348137307609780849594370286001859037097129703289653603153319926143668628405, to_address_hash: "0x2fb2ea29877371bac23a8b11ed3e373b6d380de2", transaction_index: 0, type: 2, v: 1, value: 0}, %{block_hash: "0x7f7e94411aa09e0384db618b3a7342ca9f6aa0071340965cec273959dc885873", block_number: 30063563, from_address_hash: "0xb8ff877ed78ba520ece21b1de7843a8a57ca47cb", gas: 298396, gas_price: 1320000000, hash: "0x38cd274f8af9fed180151b7708d275983eaf5947506197d24052c15a48f04751", index: 1, input: "0x6c459a2800000000000000000000000038de71124f7a447a01d67945a51edce9ff491251000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000000650b11dd00000000000000000000000000000000000000000000000000000000000001400000000000000000000000000000000000000000000000000000000000000084704316e500000000000000000000000000000000000000000000000000000000000000666d6ddfd3bf572ca788f611702caac3e92f11066a9abe8f61fccde28005413eba00000000000000000000000000000000000000000000000000000000000000146d6ddfd3bf572ca788f611702caac3e92f11066a9abe8f61fccde28005413eba000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000082509cb02eaa050fc9b5c96214fdddac1de5d7bd24b033a8aaf301997bc3a01f1f5881fda5c9d09138bf66fa511d41bc3e35d95d3ab679600e7e7a1785ea5accc31b2321f10fff24d60fca7cea3903b124ba1e18e86155f5adba905c5bf4a0a4e80d5563c300d38da28c8bc514acb7f85b031c53ff0eacbbaa30db0063a8710051ea1b000000000000000000000000000000000000000000000000000000000000", nonce: 3506778, r: 106914541084919803790845011255419019755405946954016282156034675113701718062859, s: 55930121912161762707148658847347982564618242016833021238596893501868563084763, to_address_hash: "0xa6bf2be6c60175601bf88217c75dd4b14abb5fbb", transaction_index: 1, type: 0, v: 236, value: 0}, %{block_hash: "0x7f7e94411aa09e0384db618b3a7342ca9f6aa0071340965cec273959dc885873", block_number: 30063563, from_address_hash: "0xe93685f3bba03016f02bd1828badd6195988d950", gas: 1022812, gas_price: 1320000000, hash: "0xa2c1c3c128ae6f5e7561d491b0359d28cde90e5967b492c1e146ad8a00853445", index: 2, input: "0x252f7b01000000000000000000000000000000000000000000000000000000000000006d000000000000000000000000fa5ed56a203466cbbc2430a43c66b9d8723528e70000000000000000000000000000000000000000000000000000000000030d4044ad4d1a7cce24187cf12c2d41d31b11fa6d2ece1114b9b63d58318a888b13ad44ad4d1a7cce24187cf12c2d41d31b11fa6d2ece1114b9b63d58318a888b13ad00000000000000000000000000000000000000000000000000000000000000c000000000000000000000000000000000000000000000000000000000000000d40000000000000000000000004d73adb72bc3dd368966edd0f0b2148401a178e200000000000ee07a006d0c1ebbb61374da1a8c57cb6681bf27178360d36f0091fa5ed56a203466cbbc2430a43c66b9d8723528e7000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000005af3107a4000000000000000000000000000000000000000000000000000000000000000001495c066f05650abc1fbbde992c647fa760c8a33e7000000000000000000000000000000000000000000000000", nonce: 3554277, r: 113809503637058026690056241524756506163664925068125807327478949375912999335160, s: 9803266047688982554281198374939818742844196991122801567689331723907432762184, to_address_hash: "0x5b19bd330a84c049b62d5b0fc2ba120217a18c1c", transaction_index: 2, type: 0, v: 236, value: 0}, %{block_hash: "0x7f7e94411aa09e0384db618b3a7342ca9f6aa0071340965cec273959dc885873", block_number: 30063563, from_address_hash: "0xe93685f3bba03016f02bd1828badd6195988d950", gas: 1022812, gas_price: 1320000000, hash: "0xc939f75927c93b98a9436575de8fba76e5f989ff479d51e71c03e32fc10e208b", index: 3, input: "0x252f7b01000000000000000000000000000000000000000000000000000000000000009e000000000000000000000000fa5ed56a203466cbbc2430a43c66b9d8723528e70000000000000000000000000000000000000000000000000000000000030d40efbb9b473acf18f4f94aa37b9161ce86d018c2e9a1ac6a96b957a3213a7e7aeaefbb9b473acf18f4f94aa37b9161ce86d018c2e9a1ac6a96b957a3213a7e7aea00000000000000000000000000000000000000000000000000000000000000c000000000000000000000000000000000000000000000000000000000000000d4000000000000000000000000fe7c30860d01e28371d40434806f4a8fcdd3a0980000000000000bcb009e2859a4ebcb58c8dd5cac1419c4f63a071b642b200091fa5ed56a203466cbbc2430a43c66b9d8723528e7000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000a039d318680a9f90800000000000000000000000000000000000000000000000000000000000000143e59ab9ee143e2d201fcbd884849dedb40ef5b36000000000000000000000000000000000000000000000000", nonce: 3554278, r: 109888720974429334848546672366847280355008808493039008539331176449788354621997, s: 38911446133594825689001384403520507634475434433242100542279780075381719256252, to_address_hash: "0x5b19bd330a84c049b62d5b0fc2ba120217a18c1c", transaction_index: 3, type: 0, v: 236, value: 0}, %{block_hash: "0x7f7e94411aa09e0384db618b3a7342ca9f6aa0071340965cec273959dc885873", block_number: 30063563, from_address_hash: "0x90bf74d7e819d7b6fb7d67ba292b4cf2cea5b507", gas: 132403, gas_price: 1100000000, hash: "0xb274b59ee68bd89e661cbe88a7d5eac1a9943a12e391cdb58d958a3f9860dfd4", index: 4, input: "0xde943b2900000000000000000000000000000000000000000000000000000000650abb980000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000018000000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000847b225369676e4465636f6d6d6974223a5b35362c3130332c3233382c3132392c37372c3139382c3232392c37352c39392c3233362c34342c3135362c3132352c34352c38332c3233362c3231362c3230382c32392c36352c3230382c3138392c3235322c33342c3137322c32302c3234322c36322c32372c3234352c3231342c38375d7d00 (truncated)
```
```
2023-09-20T12:38:16.492 [error] Task #PID<0.24655.1> started from Indexer.Block.Realtime.Fetcher terminating
** (FunctionClauseError) no function clause matching in anonymous fn/2 in Indexer.Block.Fetcher.Receipts.fetch/2
    (indexer 5.2.2) anonymous fn({:exit, {%FunctionClauseError{module: :prim_inet, function: :recv0, arity: 3, kind: nil, args: nil, clauses: nil}, [{:prim_inet, :recv0, [#Port<0.125>, -1, 600000], []}, {:hackney_response, :stream_body_recv, 2, [file: '/Users/nikitosing/work/blockscout/blockscout/deps/hackney/src/hackney_response.erl', line: 170]}, {:hackney_response, :read_body, 3, [file: '/Users/nikitosing/work/blockscout/blockscout/deps/hackney/src/hackney_response.erl', line: 312]}, {:hackney, :"-body/2-fun-0-", 2, [file: '/Users/nikitosing/work/blockscout/blockscout/deps/hackney/src/hackney.erl', line: 520]}, {HTTPoison.Base, :request, 6, [file: 'lib/httpoison/base.ex', line: 888]}, {EthereumJSONRPC.HTTP.HTTPoison, :json_rpc, 4, [file: 'lib/ethereum_jsonrpc/http/httpoison.ex', line: 12]}, {EthereumJSONRPC.HTTP, :chunked_json_rpc, 3, [file: 'lib/ethereum_jsonrpc/http.ex', line: 73]}, {EthereumJSONRPC.RequestCoordinator, :"-perform/4-fun-0-", 3, [file: 'lib/ethereum_jsonrpc/request_coordinator.ex', line: 87]}, {EthereumJSONRPC.RequestCoordinator, :trace_request, 2, [file: 'lib/ethereum_jsonrpc/request_coordinator.ex', line: 107]}, {EthereumJSONRPC, :json_rpc, 2, [file: 'lib/ethereum_jsonrpc.ex', line: 441]}, {EthereumJSONRPC.Receipts, :fetch, 2, [file: 'lib/ethereum_jsonrpc/receipts.ex', line: 136]}, {Task.Supervised, :invoke_mfa, 2, [file: 'lib/task/supervised.ex', line: 89]}, {Task.Supervised, :reply, 4, [file: 'lib/task/supervised.ex', line: 34]}, {:proc_lib, :init_p_do_apply, 3, [file: 'proc_lib.erl', line: 240]}]}}, {:ok, %{logs: [], receipts: []}}) in Indexer.Block.Fetcher.Receipts.fetch/2
    (elixir 1.14.5) lib/task/supervised.ex:370: Task.Supervised.stream_deliver/7
    (elixir 1.14.5) lib/enum.ex:2514: Enum.reduce_while/3
    (indexer 5.2.2) lib/indexer/block/fetcher/receipts.ex:22: Indexer.Block.Fetcher.Receipts.fetch/2
    (indexer 5.2.2) lib/indexer/block/fetcher.ex:138: Indexer.Block.Fetcher.fetch_and_import_range/2
    (stdlib 4.3.1.2) timer.erl:235: :timer.tc/1
    (indexer 5.2.2) lib/indexer/block/realtime/fetcher.ex:307: Indexer.Block.Realtime.Fetcher.do_fetch_and_import_block/3
    (indexer 5.2.2) lib/indexer/logger.ex:14: Indexer.Logger.metadata/2
    (indexer 5.2.2) lib/indexer/block/realtime/fetcher.ex:287: Indexer.Block.Realtime.Fetcher.fetch_and_import_block/4
    (elixir 1.14.5) lib/task/supervised.ex:89: Task.Supervised.invoke_mfa/2
    (stdlib 4.3.1.2) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
Function: &Indexer.Block.Realtime.Fetcher.fetch_and_import_block/3
    Args: [30063563, %Indexer.Block.Fetcher{broadcast: :realtime, callback_module: Indexer.Block.Realtime.Fetcher, json_rpc_named_arguments: [transport: EthereumJSONRPC.HTTP, else: EthereumJSONRPC.IPC, transport_options: [http: EthereumJSONRPC.HTTP.HTTPoison, url: "...", fallback_url: nil, fallback_trace_url: nil, method_to_url: [eth_getBalance: "...", trace_block: "...", trace_replayBlockTransactions: "...", trace_replayTransaction: "..."], http_options: [recv_timeout: 600000, timeout: 600000, hackney: [pool: :ethereum_jsonrpc]]], variant: EthereumJSONRPC.Nethermind], receipts_batch_size: 250, receipts_concurrency: 10}, false]
```

**Comment by vbaranov** [comment]
@nikitosing please check with update to [Hackney 1.19.1](https://github.com/blockscout/blockscout/pull/8510). Does the issue appear after update?

**Comment by nikitosing** [comment]
@vbaranov it looks like the issues are fixed with `1.19.1`

**Comment by vbaranov** [comment]
Resolved with https://github.com/blockscout/blockscout/pull/8510
